pyanaconda/modules/storage/partitioning/automatic/automatic_partitioning.py

```diff
@@ -122,37 +123,59 @@ def _get_luks_format_args(storage, request):
         }
 
     @staticmethod
-    def _get_partitioning(storage, scheme, excluded_mount_points=()):
+    def _get_partitioning(storage, scheme, request: PartitioningRequest):
         """Get the partitioning requests for autopart.
 
         :param storage: blivet.Blivet instance
         :param scheme: a type of the partitioning scheme
-        :param excluded_mount_points: a list of mount points to exclude
+        :param request: partitioning parameters
         :return: a list of full partitioning specs
         """
-        requests = []
+        specs = []
+        will_swap_be_created = False
 
-        for request in get_default_partitioning():
+        for spec in get_default_partitioning():
             # Skip mount points excluded from the chosen scheme.
-            if request.schemes and scheme not in request.schemes:
+            if spec.schemes and scheme not in spec.schemes:
                 continue
 
             # Skip excluded mount points.
-            if (request.mountpoint or request.fstype) in excluded_mount_points:
+            if (spec.mountpoint or spec.fstype) in request.excluded_mount_points:
                 continue
 
             # Set the default filesystem type.
-            if request.fstype is None:
-                request.fstype = storage.get_fstype(request.mountpoint)
+            if spec.fstype is None:
+                spec.fstype = storage.get_fstype(spec.mountpoint)
 
             # Update the size of swap.
-            if request.fstype == "swap":
+            if spec.fstype == "swap":
+                will_swap_be_created = True
                 disk_space = storage.get_disk_free_space()
-                request.size = suggest_swap_size(disk_space=disk_space)
-
-            requests.append(request)
-
-        return requests
+                spec.size = suggest_swap_size(hibernation=request.hibernation,
+                                              disk_space=disk_space)
+
+            specs.append(spec)
+
+        # When request.hibernation is set swap partition should be created even if it wasn't
+        # included in the default partitioning scheme.
+        if not will_swap_be_created and request.hibernation:
+            disk_space = storage.get_disk_free_space()
+            spec = PartSpec(
```

Comment:
This is dangerous. If we ever change the `get_default_partitioning` function, we can easily forget to update this part of the code. These bugs are tricky to find. You will have to figure out how to reuse the code of the `get_default_partitioning` function and probably create a helper function that converts `attrs` to `PartSpecs`.